components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1039,8 +1056,23 @@ void RewardsServiceImpl::TriggerOnRecoverWallet(ledger::Result result,
 }
 
 void RewardsServiceImpl::SolveGrantCaptcha(const std::string& solution) const {
+#if !defined(OS_ANDROID)
   return ledger_->SolveGrantCaptcha(solution);
+#else
+  safetynet_check::ClientAttestationCallback attest_callback = base::BindOnce(&RewardsServiceImpl::GrantAttestationResult, base::Unretained(this));
+  safetynet_check_runner_.performSafetynetCheck(solution, std::move(attest_callback));
+#endif
+}
+
+#if defined(OS_ANDROID)
+void RewardsServiceImpl::GrantAttestationResult(bool result, const std::string& result_string) const {
+  if (result) {
+    return ledger_->SolveGrantCaptcha(result_string, is_android_);
+  } else {
+    LOG(ERROR) << "GrantAttestationResult error: " << result_string;
```

Comment:
we can't just log the error here, we need to report it back with `OnGrantCaptcha`